src/LaraFormsBuilder.php

```diff
@@ -496,4 +508,27 @@ protected function getDefaultCardFieldErrorWrapperClasses(): string
     {
         return config('lara-forms-builder.card_field_error_wrapper_classes');
     }
+
+    protected function extraCheckRequiredField($fieldKey): bool
+    {
+        return false;
+    }
+
+    /**
+     * Check if the field is required
+     *
+     * @param  string  $fieldKey
+     */
+    protected function isFieldRequired($fieldKey): bool
+    {
+        return
+            isset($this->rules)
+            && array_key_exists('formProperties.'.$fieldKey, $this->rules)
+            && ((
+                str_contains($this->rules['formProperties.'.$fieldKey], 'required')
+                && ! str_contains($this->rules['formProperties.'.$fieldKey], 'required_')
+            )
+            || $this->extraCheckRequiredField($fieldKey)
```

Comment:
Why is this only called when there are rules defined for the field key? I can imagine cases where you want to have only dynamic conditions. If we keep it that way, it should be mentioned as a prerequisite condition in the README documentation of `extraCheckRequiredField`